src/Features/VisualBasic/Portable/UseInferredMemberName/VisualBasicUseInferredMemberNameDiagnosticAnalyzer.vb

```diff
@@ -0,0 +1,109 @@
+
+            Dim syntaxTree = context.Node.SyntaxTree
+            Dim optionSet = context.Options.GetDocumentOptionSetAsync(syntaxTree, cancellationToken).GetAwaiter().GetResult()
+            If optionSet Is Nothing Then
+                Return
+            End If
+
+            Dim parseOptions = DirectCast(syntaxTree.Options, VisualBasicParseOptions)
+            Select Case context.Node.Kind()
+                Case SyntaxKind.NameColonEquals
+                    ReportDiagnosticsIfNeeded(DirectCast(context.Node, NameColonEqualsSyntax), context, optionSet, syntaxTree)
+                    Exit Select
+                Case SyntaxKind.NamedFieldInitializer
+                    ReportDiagnosticsIfNeeded(DirectCast(context.Node, NamedFieldInitializerSyntax), context, optionSet, syntaxTree)
+                    Exit Select
+            End Select
+        End Sub
+
+        Private Function ReportDiagnosticsIfNeeded(nameColonEquals As NameColonEqualsSyntax, context As SyntaxNodeAnalysisContext,
+                                                   optionSet As OptionSet, syntaxTree As SyntaxTree) As Boolean
+
+            If Not nameColonEquals.IsParentKind(SyntaxKind.SimpleArgument) Then
+                Return False
+            End If
+
+            Dim argument = DirectCast(nameColonEquals.Parent, SimpleArgumentSyntax)
+            If Not VisualBasicInferredMemberNameReducer.CanSimplifyTupleName(argument) Then
+                Return False
+            End If
+
+            ' Create a normal diagnostic
+            context.ReportDiagnostic(
+                Diagnostic.Create(GetDescriptorWithSeverity(
+                    optionSet.GetOption(VisualBasicCodeStyleOptions.PreferInferredTupleNames).Notification.Value),
+                    nameColonEquals.GetLocation()))
+
+            ' Also fade out the part of the name-colon-equals syntax
+            Dim fadeSpan = TextSpan.FromBounds(nameColonEquals.Name.SpanStart, nameColonEquals.ColonEqualsToken.Span.End)
+            context.ReportDiagnostic(
+                Diagnostic.Create(
+                    UnnecessaryWithoutSuggestionDescriptor,
+                    syntaxTree.GetLocation(fadeSpan)))
+
+            Return True
+        End Function
+
+        Private Function ReportDiagnosticsIfNeeded(fieldInitializer As NamedFieldInitializerSyntax, context As SyntaxNodeAnalysisContext,
+                                                   optionSet As OptionSet, syntaxTree As SyntaxTree) As Boolean
+
+            If Not VisualBasicInferredMemberNameReducer.CanSimplifyNamedFieldInitializer(fieldInitializer) Then
+                Return False
+            End If
+
+            ' Create a normal diagnostic
+            context.ReportDiagnostic(
+                Diagnostic.Create(GetDescriptorWithSeverity(
+                    optionSet.GetOption(VisualBasicCodeStyleOptions.PreferInferredAnonymousTypeMemberNames).Notification.Value),
+                    fieldInitializer.GetLocation()))
```

Comment:
is this the span of the entire "x = y"?  if so, this i likely too large as y could be arbitrarily large.  (same comment for C# if it applies).